### ISSUE TITLE
About the semantic loss

### ISSUE THREAD

**Issue by sjtuplayer** [showlab/FQGAN#1]
I find that the semantic loss has no grad. So, what is its function? Is it the Rep. Learning Loss mentioned in the paper?
https://github.com/showlab/FQGAN/blob/a1265f2002eeea4750dbb8bea7908c72eb86923a/tokenizer/models/clip_encoder.py#L88

**Comment by JosephPai** [comment]
Hi @sjtuplayer , thanks for the question. This is a nice catch.

In the code we used for running experiments, we don't really have this line. While before releasing the code, we add this line to double-confirm that the CLIP encoder will not trained.
In fact, in the loss definition, we have explicitly set requires_grad_ = False [(Code)](https://github.com/showlab/FQGAN/blob/a1265f2002eeea4750dbb8bea7908c72eb86923a/tokenizer/vq_loss.py#L249), this already ensures that the parameters of CLIP encoder will not appear in the optimizer.

So, please feel free to comment out this line, as this is duplicated and may cause potential issue in training. We will also update the code.

**Comment by sjtuplayer** [comment]
> Hi [@sjtuplayer](https://github.com/sjtuplayer) , thanks for the question. This is a nice catch.
> 
> In the code we used for running experiments, we don't really have this line. While before releasing the code, we add this line to double-confirm that the CLIP encoder will not trained. In fact, in the loss definition, we have explicitly set requires_grad_ = False [(Code)](https://github.com/showlab/FQGAN/blob/a1265f2002eeea4750dbb8bea7908c72eb86923a/tokenizer/vq_loss.py#L249), this already ensures that the parameters of CLIP encoder will not appear in the optimizer.
> 
> So, please feel free to comment out this line, as this is duplicated and may cause potential issue in training. We will also update the code.

Thanks for your reply. So, if the mentioned line is not really used, where is the semantic loss calculated? And is it the Rep. Learning Loss mentioned in the paper? If not, where is the Rep. Learning Loss? I'm quite curious about this loss since the paper does not have details for it. Looking forward for your response!

**Comment by JosephPai** [comment]
https://github.com/showlab/FQGAN/blob/main/tokenizer/models/clip_encoder.py#L88

This exactly how we compute the semantic loss.
"the mentioned line is not really used" means that we don't really need to add `torch.no_grad()` here.

**Comment by sjtuplayer** [comment]
> https://github.com/showlab/FQGAN/blob/main/tokenizer/models/clip_encoder.py#L88
> 
> This exactly how we compute the semantic loss. "the mentioned line is not really used" means that we don't really need to add `torch.no_grad()` here.

Thanks a lot.
